platform/openide.actions/src/org/openide/actions/HeapView.java

```diff
@@ -465,526 +225,174 @@ private void updateTextWidth() {
+            g2.fillRect(0, 0, width + 1, height + 1);
+            // Draw samples
+            g2.setPaint(CHART_COLOR);
+            paintSamples(g2, width, height);
+            // Draw text if enabled
             if (getShowText()) {
-                if (getShowDropShadow()) {
-                    paintDropShadowText(g, innerW, innerH);
-                } else {
-                    g.setColor(textColor);
-                    paintText(g, innerW, innerH);
-                }
+                paintText(g2, width, height);
             }
-            g.translate(-1, -2);
         } else {
             stopTimerIfNecessary();
-            // To honor opaque contract, fill in the background
-            g.setColor(getBackground());
-            g.fillRect(0, 0, width, height);
-        }
-    }
-    
-    private void paintTicks(Graphics2D g, int width, int height) {
-        if (graphIndex > 0 || graphFilled) {
-            int index = getGraphStartIndex();
-            int x = 0;
-            if (!graphFilled) {
-                x = width - graphIndex;
-            }
-            float[] localGraph = graph;
-            if (localGraph == null) {
-                return;
-            }
-            float min = localGraph[index];
-            index = (index + 1) % localGraph.length;
-            while (index != graphIndex) {
-                min = Math.min(min, localGraph[index]);
-                index = (index + 1) % localGraph.length;
-            }
-            int minHeight = (int)(min * (float)height);
-            if (minHeight > 0) {
-               g.drawImage(tickGradientImage, x, height - minHeight, width, height,
-                        x, height - minHeight, width, height, null);
-            }
-            index = getGraphStartIndex();
-            do {
-                int tickHeight = (int)(localGraph[index] * (float)height);
-                if (tickHeight > minHeight) {
-                    g.drawImage(tickGradientImage, x, height - tickHeight, x + 1, height - minHeight,
-                            x, height - tickHeight, x + 1, height - minHeight, null);
-                }
-                index = (index + 1) % localGraph.length;
-                x++;
-            } while (index != graphIndex);
         }
     }
 
     /**
-     * Renders the text.
+     * Renders the text using an optional drop shadow.
      */
-    private void paintText(Graphics g, int w, int h) {
+    private void paintText(Graphics2D g, int w, int h) {
         g.setFont(getFont());
         String text = getHeapSizeText();
         FontMetrics fm = g.getFontMetrics();
         int textWidth = fm.stringWidth(text);
-        g.drawString(text, (w - maxTextWidth) / 2 + (maxTextWidth - textWidth), 
-                h / 2 + fm.getAscent() / 2 - 2);
+        int x = (w - maxTextWidth) / 2 + (maxTextWidth - textWidth);
+        int y = h / 2 + fm.getAscent() / 2 - 2;
+        g.setColor(TEXT_COLOR);
+        g.drawString(text, x, y);
```

Comment:
It _was_ me.. I had that and took it out because it didn't seem to add much to the clarity in the original scheme. Here it is with/without drop shadow, using your theme colours of 0x2e90e8/0xCEDBE6
![Screenshot 2020-02-04 at 20 42 55](https://user-images.githubusercontent.com/22172519/73785462-38610800-478f-11ea-8755-3ec6ed1066ba.png)
![Screenshot 2020-02-04 at 20 41 51](https://user-images.githubusercontent.com/22172519/73785463-38610800-478f-11ea-8bb5-487dd966b3e1.png)
I think there is insufficient contrast between white text and the pale blue, even with the drop shadow. May be better to go black text, what do you think?

Reply:
I think black is too much, makes it look more important than all the other buttons on the ribbon.
![Screenshot 2020-02-04 at 20 53 05](https://user-images.githubusercontent.com/22172519/73786779-df46a380-4791-11ea-9a0f-8685aa78ee9d.png)

How about `#404040`?
![Screenshot 2020-02-04 at 20 58 14](https://user-images.githubusercontent.com/22172519/73786777-deae0d00-4791-11ea-8ab6-0f807e26a191.png)
Or even not bold?
![Screenshot 2020-02-04 at 20 54 45](https://user-images.githubusercontent.com/22172519/73786780-df46a380-4791-11ea-985b-33e89b20d349.png)
Sorry for bike shedding!

Reply:
Perhaps try black at 85% alpha. Would be good to take out the bold, too, yes.

I also experimented (in Photoshop) with adding a semitransparent dark border (black at 25% alpha) to make the background go better with the surroundings. See screenshot. Though it's a little tricky to paint borders that don't become blurry on fractional HiDPI scalings (o.n.swing.plaf/src/org/netbeans/swing/plaf/windows8/DPIUnscaledBorder.java  shows an example), so that can be postponed...

![memory](https://user-images.githubusercontent.com/886243/73787489-6c293700-4769-11ea-9c6b-cbcec27da417.png)

Reply:
I did some more experiments and it seemed that the dark gray was legible without being too prominent. Rendering with alpha didn't seem to make it better. Agree on the border looking good but please can we defer this for the next version!

Reply:
Yes, happy to defer further improvements! I might look into adding a DPI-safe border myself at some point, after these bigger changes are merged.